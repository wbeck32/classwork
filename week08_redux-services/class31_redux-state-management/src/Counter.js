import React from 'react';
import { connect } from 'react-redux';
import { increment, decrement } from './counter.actions';

export function Counter({ count, dispatch }) {
  return (
    <div>
      <button 
        onClick={() => dispatch(decrement())}
        disabled={count <= 0}
      >-</button>
      <span style={{ padding: '10px' }}>
        {count}
      </span>
      <button onClick={() => dispatch(increment())}>+</button>
    </div>
  );
}

export default connect(
  state => ({ count: state.count })
)(Counter);