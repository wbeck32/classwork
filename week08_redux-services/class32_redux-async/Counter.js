import React, { Component } from 'react';

class CounterContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      count: 0
    }
  }

  changeCounter(val) {
    this.setState({ count: this.state.count + val });
  }

  render() {
    const { count } = this.state;
    return <Counter count={count} 
      incr={() => this.changeCounter(-1)}
      decr={() => this.changeCounter(+1)}/>;
  }
}

function Counter({ count, incr, decr }) {
  return (
    <div>
      <button onClick={() => incr()}>-</button>
      <span>{count}</span>
      <button onClick={() => decr()}>+</button>
    </div>
  );
}