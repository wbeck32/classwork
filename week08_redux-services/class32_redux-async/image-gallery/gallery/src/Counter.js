import React, { Component } from 'react';

export default class CounterContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
  }

  changeCounter(val) {
    this.setState((banana, props) => {
      return { count: banana.count + val };
    });
    
    this.setState(({ count }) => {
      return { count: count + val };
    });

  }

  render() {
    const { count } = this.state;
    return <Counter count={count} 
      incr={() => this.changeCounter(-1)}
      decr={() => this.changeCounter(+1)}/>;
  }
}

export function Counter({ count, incr, decr }) {
  return (
    <div>
      <button onClick={() => incr()}>-</button>
      <span>{count}</span>
      <button onClick={() => decr()}>+</button>
    </div>
  );
}