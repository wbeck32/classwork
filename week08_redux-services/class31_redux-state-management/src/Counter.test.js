import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Counter } from './Counter';

describe('Counter', () => {

  it('shows count and two buttons', () => {
    const wrapper = shallow(<Counter count={3}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('disables decrement when count is 0', () => {
    const wrapper = shallow(<Counter count={0}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});