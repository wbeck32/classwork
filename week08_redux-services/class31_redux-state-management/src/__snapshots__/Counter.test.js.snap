// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter disables decrement when count is 0 1`] = `
<div>
  <button
    disabled={true}
    onClick={[Function]}
  >
    -
  </button>
  <span
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    0
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`Counter shows count and two buttons 1`] = `
<div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    -
  </button>
  <span
    style={
      Object {
        "padding": "10px",
      }
    }
  >
    3
  </span>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
